Add prune until filter test for podman volume cli

jmguzik opened the pull request:
This commit follows work started in #10756. Changes made in #11015
enabled cli support for volume prune --filter until. Adding e2e test
 closes #10579.

Fixes #10579

jmguzik commented on pkg/domain/infra/abi/volumes.go:
```diff
@@ -154,6 +155,24 @@ func (ic *ContainerEngine) VolumeList(ctx context.Context, opts entities.VolumeL
+	)
+	// special internal request handler to get all until=time
+	// volumes using prune filters instead of list filters
+	if opts.UntilCmdRequest {
```
Added this one to not expose until in list filters portfolio and keep the current architecture (internal request possibility) in cmd volumes prune. I personally think there is no better way to do this currently, but I am open to suggestions.

Luap99 replied:
Is there any reason why until should not be allowed for list?

jmguzik replied:
@Luap99 yes. It is an extension as in docker it does not exist. If you accept this as compat extension or compatibility breakage it's fine. If not, we should stick with the proposed solution. Let me know what you think.

Luap99 replied:
I don't see any problems adding until to list. Adding more filters is not a compatibility breakage for me. I mean who is going to call list with the until filter and expect it to fail.
But let's wait what other maintainers think, @rhatdan @mheon @vrothberg?

jmguzik replied:
Not a problem @Luap99. Let's hear other comments and if so, i can add one more filter. I just followed docker example here.

vrothberg replied:
I am cool if the compat layer can do more than Docker.

mheon replied:
Yeah, our big concern is that we don't do less than they do, and what we both do should match Docker's behavior closely. If our compat code supports extra options, it's not a big issue.

jmguzik replied:
@mheon @Luap99 @vrothberg https://github.com/containers/podman/pull/11015 was created as prerequisite to this PR

jmguzik replied:
Done

jmguzik commented:
I think CI failure is not related.

vrothberg commented on pkg/domain/entities/volumes.go:
```diff
@@ -114,7 +114,8 @@ type VolumePruneOptions struct {
 type VolumeListOptions struct {
-	Filter map[string][]string
+	Filter          map[string][]string
+	UntilCmdRequest bool
```
Why do we need an extra field?  I'd expect "until" to be a key in the Filter map.

jmguzik replied:
If you are cool then we do not need this field. I will correct it.

jmguzik replied:
Done

edsantiago commented on test/e2e/volume_prune_test.go:
```diff
@@ -63,6 +63,37 @@ var _ = Describe("Podman volume prune", func() {
+	It("podman prune volume --filter until", func() {
+		session := podmanTest.Podman([]string{"volume", "create", "--label", "label1=value1", "myvol1"})
+		session.WaitWithDefaultTimeout()
+		Expect(session.ExitCode()).To(Equal(0))
```
This pattern is anti-recommended, please see #10997. Could you fix it (and all instances) to use the recommended form (`Expect(session).To(Exit(0))`)? Thank you!

jmguzik replied:
Ok, will fix

jmguzik replied:
Done

jmguzik commented:
It seems only e2e test is needed to make it working after changes made in #11015. I changed the commit and PR description. Adjusted it also to your suggestions.

edsantiago commented:
LGTM. It bothers me that `until` is not documented in `podman-volume-prune(1)`, but that is beyond the scope of your PR.

jmguzik commented:
> LGTM. It bothers me that `until` is not documented in `podman-volume-prune(1)`, but that is beyond the scope of your PR.

I will add it. I should do it earlier, forgot so I will do it now.

edsantiago commented:
Looks like you're also missing a `WaitWithDefaultTimeout()`

jmguzik commented:
> Looks like you're also missing a `WaitWithDefaultTimeout()`

Yes, added. I also reviewed the situation in podman-volume-prune(1). Most of the filters were wrong for this command according to:
https://docs.podman.io/en/latest/_static/api.html#operation/VolumePruneLibpod

I updated the info. Good catch.

edsantiago commented:
/lgtm
